### PR TITLE
refactor(core): remove vestigial deprecation path

### LINKED ISSUE #175: Phase 0: remove the vestigial deprecation path
## Goal

Remove the vestigial v5 deprecation helper, its self-only tests, and the stale `DEV_MODE` feature documentation instead of preserving dead source as an implied contract.

## Agent-development failure addressed

`utils/deprecate.js` appears to be supported framework functionality because it has a large dedicated test suite and current documentation claims `DEV_MODE` emits warnings. In fact, no production source imports it, it is absent from every shipped entry graph, it is neither exported nor packed, and no runtime path can emit those warnings. Agents should not spend time understanding or extending a test-preserved dead path.

## Public behavior contract

The v5 public package does not expose a deprecation helper or promise `DEV_MODE` warnings. Future deprecations must enter with a real caller and the stable diagnostic-catalog contract. Historical v3-to-v4 migration documentation remains unchanged.

## Runtime cost category

Removes an unused feature-map entry; expected runtime and bundle impact is neutral or a small reduction. No fallback, alias, or compatibility path is retained because there is no verified v5 consumer.

## Acceptance criteria

- [ ] Delete `utils/deprecate.js` and its isolated unit test.
- [ ] Remove the unused initial `DEV_MODE` feature-map entry.
- [ ] Remove the stale current documentation claim that `DEV_MODE` emits runtime warnings.
- [ ] Preserve the historical v3-to-v4 record while clearly marking the v5 removal and recording it in the v4-to-v5 compatibility ledger.
- [ ] Do not add the dead helper to native-source guards or expose a replacement API.
- [ ] Regenerate committed artifacts and prove no public package entrypoint or packed-file contract expands.
- [ ] Record the exact executable bundle-size delta.

## Tests and evidence

Run lint, full configured coverage, documentation build/link validation, package fixtures, artifact reproduction, size reporting, and package dry-run comparison. Verify repository search finds no remaining current `DEV_MODE` or deprecation-helper contract outside historical migration text.

## Rollback or deprecation condition

Reintroduce deprecation support only with a verified public need, an actual runtime caller, and cataloged stable diagnostics. Do not restore the arbitrary-string helper as unused infrastructure.

### PR DESCRIPTION
**Pull request by paulfalgout**
Closes #175

## Linked issue

- Closes #175

## What

- Delete the unreferenced `utils/deprecate.js` helper and its self-only unit tests.
- Remove the unused initial `DEV_MODE` feature key and the current documentation claim that it emits warnings.
- Regenerate the executable bundles and sourcemaps from the canonical source.

## Agent-development failure addressed

The helper, its 10 dedicated tests, and current `DEV_MODE` documentation made a dead private path look like supported framework behavior. No production source imported it, and it was neither exported nor packed. Removing it leaves agents one accurate implementation surface instead of test-preserved vestigial code.

## Public behavior contract

- Canonical behavior after this PR: v5 exposes no deprecation helper and makes no `DEV_MODE` warning promise. A future deprecation path must arrive with a real runtime caller and cataloged diagnostics.
- Superseded behavior removed, if any: the private arbitrary-string helper, its isolated tests, and the unused initial `DEV_MODE` key/current docs claim.
- Compatibility exception and removal condition, if any: none; no v5 package export, packed source path, runtime caller, or verified consumer exists.

## Runtime cost

- [ ] Static or documentation only
- [ ] Development or test only; excluded from production entrypoints
- [x] Changes an existing production path
- [ ] Adds an explicitly opt-in runtime path

Measured impact or explanation:

- Bundle: Brotli delta versus `cb9c3ca`: ES module −13 B (12,789 → 12,776 B), CommonJS −12 B (12,944 → 12,932 B), minified UMD −17 B (10,018 → 10,001 B), Backbone shim 0 B. Raw executable deltas are −19 B ESM, −19 B CJS, −21 B UMD, and −12 B minified UMD; adapter bundles are byte-identical.
- Startup/hot path: no work added; one unused property is removed from the static feature map.
- Allocations/retention: one unused static object property is removed; no new allocation or retained state.

## Validation

- [x] Tests cover the acceptance criteria and relevant edge cases.
- [x] Docs, examples, declarations, diagnostics, and rule metadata agree where relevant.
- [x] No private consumer, private fixture, or undocumented context is required.
- [x] I ran the commands listed below and am reporting their actual results.

Commands:

```sh
npm ci
npm run lint:ci
npm run coverage
npm run docs:check
npm run test:source
npm run check:diagnostics
npm run check:browser-profile
npm run test:fixtures
npm run size
npm run check:dist
npm pack --dry-run --ignore-scripts --json
```

Results:

- Node 24.19.0 / npm 11.17.0 clean install: 357 packages, 0 audit vulnerabilities.
- Coverage: 60 files and 1,116 tests passed; 100% statements, branches, functions, and lines.
- Documentation: 28 pages built; 29 HTML files and 280 internal links validated.
- Package fixtures: CJS, ESM, adapters, peer-range, no-default-export, and Vite production build passed.
- Artifact reproduction: `npm run check:dist` passed from the committed state with a clean worktree.
- Package dry-run: identical 16-file path list; unpacked size decreased by 179 B.
- Repository search: only the historical v3-to-v4 facts and the v4-to-v5 Removed ledger entry remain; no current warning-support claim exists.

## Package and browser impact

- Entry points or install fixtures affected: existing entrypoints keep the same exports and behavior; core executable bundles shrink by the measured amounts. All package fixtures pass.
- Browser contracts affected: none; no browser behavior is added or removed because the helper had no runtime caller.
- Production/dev/test module-graph impact: removes one unreachable private source leaf and its isolated test; the production graph only loses the unused initial feature-map property.

## Scope

Core feature configuration, current feature documentation, migration ledgers, generated `dist` artifacts, and the obsolete unit test are in scope. The v3-to-v4 facts are preserved while the page now points readers to the v5 removal record; package exports, diagnostic entries, and adapters are unchanged.

## Deployment Impact

No application, form, or bundle redeploy is required. The change takes effect only in a future normal Marionette package publication, and consumers do not need a coordinated migration.

## Rollback or deprecation

Reintroduce deprecation support only when a verified public need has an actual runtime caller and stable cataloged diagnostics. Do not restore the unused arbitrary-string helper.

## Review notes

The executable bundle diff is the removal of `DEV_MODE: false` from the internal feature map. Sourcemaps are regenerated reproducibly, the packed file set does not expand, and the migration pages distinguish historical v4 behavior from its v5 removal.

---
## Summary by cubic
Removes the unreferenced `utils/deprecate.js` helper, its tests, and the `DEV_MODE` feature key that documented warnings nothing ever emitted, and updates the migration docs to record the removal. Bundle sizes drop by 12–21 bytes with no runtime behavior changes.

- Adds `DEV_MODE` removal entries to `docs/migration-from-v4.md` and `docs/upgrade-v3-v4.md`; the historical v3-to-v4 section is now marked as describing that transition only.
- No migration is needed; the change takes effect in the next package publication.

<sup>Written for commit 3a8cfdee7275865c77a0663f3036dc2cc87a3ae1. Summary will update on new commits.</sup>

<a href="https://cubic.dev/pr/marionettejs/marionette/pull/176?utm_source=github" target="_blank" rel="noopener noreferrer" data-no-image-dialog="true"><picture><source media="(prefers-color-scheme: dark)" srcset="https://www.cubic.dev/buttons/review-in-cubic-dark.svg"><source media="(prefers-color-scheme: light)" srcset="https://www.cubic.dev/buttons/review-in-cubic-light.svg">Review in cubic</picture></a>

### CODE REVIEW

**Comment by paulfalgout** [comment]
Reviewed at `fe547bed` (the SHA I fetched). **Approve — good to merge.** Nothing blocking; one note for the changelog.

Deleting rather than repairing was the right call, and going after the `DEV_MODE` flag and its documentation is a better catch than the one I raised — I only flagged the module. `docs/features.md` was promising *"If `true`, deprecation console warnings are issued at runtime"*, and nothing in production ever called `deprecate`, so that promise was already false. Removing it makes the docs accurate rather than removing working behavior.

**I verified the public surface is genuinely unchanged**, since removing a documented flag is where this could have gone wrong. Comparing `master`'s bundle to this one:

```
                              master        pr176
isEnabled('DEV_MODE')         false         false
setEnabled('DEV_MODE', true)  → true        → true
isEnabled('childViewEventPrefix')  false    false
```

Identical, because `isEnabled` is `!!FEATURES[name]` — a missing key is falsy exactly like the explicit `DEV_MODE: false` was — and `setEnabled` assigns, so it still works for arbitrary names. I also checked whether the `FEATURES` object itself leaks: it's exported from `config/features.js` but **not** re-exported from the package entrypoint, so the removed key has no public surface at all.

Also verified: 1116 tests pass at 100% statements, branches, and functions, and `dist/` reproduces byte-exactly from a clean build. Removing the module and its 10 tests together is what keeps coverage whole — dropping only the tests would have left an uncovered file and broken the gate.

### One note for the changelog

`DEV_MODE` was a *documented* public feature flag with a documented default, so even though it was inert, it's a documented-surface removal rather than pure dead-code cleanup. `ROADMAP.md` explicitly permits this before stable ("Pre-releases may expose experimental APIs. Before stable, they may be changed or removed based on evidence"), and the PR body states the canonical behavior change plainly, so the process is right.

Worth making sure it lands in the v4→v5 migration ledger anyway: anyone who wrote `setEnabled('DEV_MODE', true)` against v4 docs will keep getting no warnings, which is the same as before but now also has no documentation explaining why. A one-line ledger entry saying the flag and its warning path were removed as never-implemented would close that loop.

<sub>Reviewed with Claude Code.</sub>

File: config/features.js [hunk 1]
```diff
@@ -3,8 +3,7 @@
 const FEATURES = {
   childViewEventPrefix: false,
   triggersStopPropagation: true,
-  triggersPreventDefault: true,
-  DEV_MODE: false
+  triggersPreventDefault: true
```
[the remark this answers is not in the document]
Comment (paulfalgout) [reply config/features.js:6]: Handled in docs/upgrade-v3-v4.md and docs/migration-from-v4.md: the old instruction is now explicitly historical, and the v5 ledger marks DEV_MODE warnings removed.

**Comment by paulfalgout** [comment]
Re-reviewed at `3a8cfdee` (the SHA I fetched). **Still approve.**

The ledger entry closes the loop, and catching `docs/upgrade-v3-v4.md` goes past what I flagged — I was only thinking about the v4→v5 path. That document was still instructing readers to use `setEnabled('DEV_MODE', true)` as current v4 guidance, so without the reframe someone following the v3→v4 upgrade would have been sent to a flag that no longer does anything. Adding the explicit **v5 status** line and pointing it at the ledger keeps the historical record intact while making the present state unambiguous.

The wording is also precise in a way that matters. The ledger says the call *"no longer enables warnings"* rather than saying it's removed or throws — which matches the actual behavior I verified: `setEnabled('DEV_MODE', true)` still executes and still returns `true`, it simply has no consumer. Someone auditing an old codebase won't go looking for a crash that doesn't happen.

Confirmed the only remaining `DEV_MODE` mentions are the two intentional ones — the ledger row and the historical v3→v4 section — with none left in source, `config/`, or any current-reference doc. All eight checks green.

Nothing outstanding from my side on this one.

<sub>Reviewed with Claude Code.</sub>
